src/components/Privacy/validateSetConsentOptions.js

```diff
@@ -12,27 +12,30 @@ import { GENERAL } from "../../constants/consentPurpose";
 
 export default objectOf({
   consent: arrayOf(
-    anyOf([
-      objectOf({
-        standard: literal("Adobe").required(),
-        version: literal("1.0").required(),
-        value: objectOf({
-          [GENERAL]: enumOf(IN, OUT).required()
+    anyOf(
+      [
+        objectOf({
+          standard: literal("Adobe").required(),
+          version: literal("1.0").required(),
```

Comment:
@Aaronius is 100% correct. I'm running this my Rares, I will update this thread soon.

Reply:
I see if we send an invalid `standard` for example, Konductor responds with the correct error message:

`code: "global:400"
message: "Invalid request. The value supplied for field 'standard' does not match your input schema."`

And there won't be an identity cookie on the response, which is expected. I think if we handle this error better, it should throw before the `Identity Cookie` check that we have.

Reply:
@jonsnyder To keep you in the loop:

I met with @raresvla then with @Aaronius to discuss this issue, and we agreed on the following: (I'll create a ticket with all the details)

1. If the Response has a status code in the `400 or 500` range, process the warnings and errors right after transferring the cookies (injectSendEdgeNetworkRequest.js line 81), and don't pass the response to the Components. the assumption is that those Response might contain a state update, but nothing else meaningful otherwise.

2. The change above requires us to add a new lifecycle hook: `onResponseFailure`, which should be called as we process the errors.

3. The Privacy Component should implement this new hook and read the cookies (onResponseFailure: readCookieIfQueueEmpty)

4. One last change that is not 100% related to error handling, `setConsent` should allow customers to pass an `identityMap` if they want to apply this consent to multiple IDs:

```
alloy("setConsent", {
      consent: [
        {
          standard,
          version,
          value: {
            general: generalPurpose
          }
        }
      ],
      identityMap: {
        HYP: [{
          id: "1234",
          authenticatedState: "ambiguous"
        }]
      }
    })
```